### PR TITLE
Run foreground service to prevent app being put to sleep

### PR DESCRIPTION
**Pull request by spuday90**

### CODE REVIEW

**Review by adeekshith — CHANGES_REQUESTED** [review]
Git conflicts

**Comment by adeekshith** [comment]
Notification icon missing

Screen Shot 2021-06-10 at 12 47 42 AM

**Comment by adeekshith** [comment]
- [ ] Also, not working when data is cleared (fresh install)

**Comment by hegocre** [comment]
I would also add an explanation on why they should enable the notification, and maybe even have it enabled by default so that they have the app working non-stop by default.

**Comment by adeekshith** [comment]
> I would also add an explanation on why they should enable the notification

+1 

```
Always-On Notification
Permanently show icon in the notification bar when Watomatic is enabled. Enable this if the app is not working properly in the background
```
But is the help text too long?

> maybe even have it enabled by default so that they have the app working non-stop by default.

Current plan is, it should be Off for everyone by default and if the app detects service is getting killed in the background, it  prompts the user to enable it.
For this PR, it is OFF for everyone. We can do service killed detection in another iteration.

Also in the future if we can get data regarding what devices or brands are having this issue, we can enable for those brands by default (For now, a few Xiaomi and OnePlus devices seem to have this issue).

**Comment by adeekshith** [comment]
- [ ] Clicking on Watomatic notification when we receive WhatsApp message (not persistent notification) is also opening Watomatic app instead of Whatsapp

**Comment by adeekshith** [comment]
> > But is the help text too long?
> 
> What about
> 
> ```
> Always-On notification
> Display a persistent notification to ensure Watomatic is not killed by the system
> ```
> 
> It's shorter and I think it'll do.
> 
> > For this PR, it is OFF for everyone. We can do service killed detection in another iteration.
> 
> Sounds good!

wato-settings-319

**Comment by spuday90** [comment]
>     * [ ]  Also, not working when data is cleared (fresh install)

Can you confirm this again @adeekshith

File: app/src/main/java/com/parishod/watomatic/service/KeepAliveService.java [hunk 1]
```diff
@@ -83,4 +96,10 @@ private boolean isMyServiceRunning() {
         Log.i ("isMyServiceRunning?", false+"");
         return false;
     }
+
+    private void startForeground(Service service) {
+        Log.e("DEBUG", "startForeground");
```
Comment (adeekshith) [inline app/src/main/java/com/parishod/watomatic/service/KeepAliveService.java:101]: `Log.e` is for errors right?

**Review by adeekshith — APPROVED** [review]

**Pull request merged**